# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/doc/Debugger/Debugger.Object.md
# Commit: e590eb34efc5
# Full Hash: e590eb34efc52ab109f2737167cb2aa0975a1792
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-06-01 09:43:14
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1624792: Remove Debugger interface r=tcampbell
#   
#   This patch removes the changes from bug 1554524 part 4. (Parts 1 and 3 of bug 1554524 were bytecode emitter cleanups that don't need to be reverted; part 2 was abandoned.)
#   
#   Depends on D116308
# ==============================================================================

diff -r 135a0e6af641 -r e590eb34efc5 js/src/doc/Debugger/Debugger.Object.md
--- a/js/src/doc/Debugger/Debugger.Object.md	Tue Jun 01 01:31:21 2021 +0000
+++ b/js/src/doc/Debugger/Debugger.Object.md	Tue Jun 01 01:31:21 2021 +0000
@@ -197,7 +197,7 @@
 
 ### `isClassConstructor`
 If the referent is a debuggee function, returns `true` if the referent is a class,
-or false if it is some other sort of function. If the referent is not a debuggee 
+or false if it is some other sort of function. If the referent is not a debuggee
 function, or not a function at all, this is `undefined`. (This is always equal to
 `obj.script.isClassConstructor`, assuming `obj.script` is a `Debugger.Script`.)
 
@@ -605,48 +605,6 @@
 If <i>binding</i> is in an uninitialized state initialize it to undefined
 and return true, otherwise do nothing and return false.
 
-### `setInstrumentation(callback, kinds)`
-If the referent is a global object, this specifies how instrumentation
-should be installed on scripts created in the future in this global's realm.
-If the referent is not a global object, throw a `TypeError`.  If the global
-already has instrumentation specified, throw an `Error`. `callback` is a
-`Debugger.Object` wrapping the callback to invoke, and `kinds` is an array
-of strings specifying the kinds of operations at which the callback should
-be invoked. Instrumentation is initially inactive, and can be activated via
-`DebuggerObject.setInstrumentationActive`. When instrumentation is active
-and an operation described by one of the instrumentation kinds executes,
-the callback is invoked with at least three arguments: the string kind of
-operation executing, the ID for the script specified by
-`Debugger.Script.setInstrumentationId`, and the bytecode offset
-of the script location following the instrumentation.  More call arguments
-are possible for some instrumentation kinds.  If an instrumented script is
-invoked without having `setInstrumentationId` called on it, it will throw an
-`Error`.  The possible instrumentation kinds and any extra arguments are as
-follows:
-
-* `main`: The main entry point of a script.
-* `entry`: Points other than the main entry point where a frame for the
-  script might begin executing when it was not previously on the stack.
-  Only applies to generator/async scripts.
-* `exit`: Points at which a script's frame will be popped or suspended.
-* `breakpoint`: Breakpoint sites in a script.
-* `getProperty`: Property read operations, including `x.f` and
-  destructuring operations. The callback will be additionally invoked with
-  the object and property name.
-* `setProperty`: Property write operations. The callback will be
-  additionally invoked with the object, property name, and rhs.
-* `getElement`: Element read operations. The callback will be additionally
-  invoked with the object and element value.
-* `setElement`: Element write operations. The callback will be
-  additionally invoked with the object, element value, and rhs.
-
-### `setInstrumentationActive(active)`
-If the referent is a global object, set whether instrumentation is active
-in the global's realm.  The instrumentation callback is only invoked when
-instrumentation is active, and code will run faster when instrumentation is
-inactive. If the referent is not a global object, throw a `TypeError`.
-If the referent has not had instrumentation installed, throw an `Error`.
-
 ### `getPromiseReactions`
 
 If the referent is a [`Promise`][promise] or a cross-compartment wrapper of one,